app/src/main/java/com/randomappsinc/studentpicker/database/NameListDataManager.java

```diff
@@ -64,6 +73,18 @@ public void deleteName(Context context, String name, int amount, String listName
         }
     }
 
+    public void setNameAmount(String name, int newAmount, int currentAmount, Context context, String listName) {
+        DataSource dataSource = new DataSource(context);
+        if (currentAmount > newAmount) {
+            int amountToDelete = currentAmount - newAmount;
+            dataSource.removeNames(name, listName, amountToDelete);
+        } else {
+            int amountToAdd = newAmount - currentAmount;
+            dataSource.addNames(name, listName, amountToAdd);
+        }
+        adjustAmountListener.onAdjustNameAmount(name, newAmount);
```

Comment:
We need to invoke onNameAdded() and onNameDeleted() in all the listeners. This is because changes to the edit list page need to be reflected in the name choice page (imagine deleting a name and then being able to choose it). This is the fastest and best experience for the user but leads to complexity for us unfortunately.